### PR TITLE
chore(55): cleaned up package.json

### LINKED ISSUE #55: Upgrading the packages to latest
### Issue Description
All the packages either have some sort of major update to them and security vulnerability

####Scope of the issue  
- Updating packages to the latest version 
- Ensuring all tests are working fine
- Nothing Breaks due to the upgrade 

### Severity/Priority
 - [ ] Critical
 - [x] High
 - [ ] Medium
 - [ ] Low

### Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.
- [x] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by ankushdharkar**
Date: Aug 21, 2023
Developer Name: Ankush Dharkar

---

## Issue Ticket Number
- Closes #55 

## Description
Fixed some typos and removed unnecessary stuff from package.json. Tweaked it up a bit too.

### Documentation Updated?

- [ ] Yes
- [x] No - Nothing to update
 
### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [x] Yes - Potential side-effects. But this is a learning repo and not used in prod, so we should be good.
- [ ] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
-    "t @testing-library/react": "16.0.0",
+    "@testing-library/dom": "^10.4.0",
+    "@testing-library/jest-dom": "^6.4.8",
+    "@testing-library/react": "^16.0.0",
     "@testing-library/user-event": "14.5.2",
-    "react": "18.3.1",
-    "react-dom": "18.3.1",
-    "react-scripts": "5.0.1",
-    "web-vitals": "4.2.3"
+    "react": "^18.3.1",
```
Comment (iamitprakash) [inline package.json:10]: can we remove ^ from all over

**Review by iamitprakash — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -39,8 +39,5 @@
   "volta": {
     "node": "18.12.1",
     "yarn": "1.22.19"
-  },
-  "devDependencies": {
-    "@babel/plugin-proposal-private-property-in-object": "7.21.11"
   }
```
Comment (shubhdevelop) [inline package.json:45]: while building this was suggested 
Screenshot from 2024-08-18 19-09-12

**Review by shubhdevelop — COMMENTED** [review]
Not sure about the devDependency, wasn't causing any break in build nor the development server, but might cause future breaks

File: package.json (hunk 1 above)
Comment (ankushdharkar) [reply package.json:10]: Why?
We want the upgraded minor version updates when we push the next set of dependencies upgrade. Don't we?

File: package.json (hunk 2 above)
Comment (ankushdharkar) [reply package.json:45]: Great catch @shubhdevelop! But even this has been deprecated. 

We need another one: https://www.npmjs.com/package/@babel/plugin-proposal-private-property-in-object

Comment (shubhdevelop) [reply package.json:45]: Sure, I'll check for the compatibility, and try to figure if this fixes it

Comment (shubhdevelop) [reply package.json:45]: [@babel/plugin-transform-private-property-in-object](https://babeljs.io/docs/babel-plugin-transform-private-property-in-object)

This package does fix the issue. 

image

**Review by prakashchoudhary07 — DISMISSED** [review]

File: package.json [hunk 3]
```diff
@@ -1,16 +1,16 @@
 {
   "name": "react-sign-up-with-tests",
-  "version": "0.1.0",
-  "private": true,
+  "version": "0.2.0",
```
Comment (prakashchoudhary07) [inline package.json:3]: Nice, we are starting versioning.
Should we also add release tags?

File: package.json (hunk 2 above)
Comment (ankushdharkar) [reply package.json:45]: Added in review changes commit e8845a384b469b723838ad795f04ddf744e2b084

**Review by shubhdevelop — APPROVED** [review]
Looks all good

**Review by shubhdevelop — APPROVED** [review]
Approved

File: package.json (hunk 1 above)
Comment (iamitprakash) [reply package.json:10]: fair enough, usually I don't take this risk as well

**Review by iamitprakash — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -41,6 +41,6 @@
     "yarn": "1.22.19"
```
Comment (prakashchoudhary07) [inline package.json:41]: NIT: We should upgrade the yarn and node versions also

Comment (ankushdharkar) [reply package.json:41]: Maybe in another PR. DX team will take care of it ;)

**Pull request merged**